test/e2e/specs/updates/keyrotation/keyrotation.go

```diff
@@ -0,0 +1,147 @@
+var _ = Describe("Key Rotation E2E tests [KeyRotation] [Update]", func() {
+	defer GinkgoRecover()
+
+	var (
+		cli        *azure.Client
+		manifest   = flag.String("manifest", "../../_data/manifest.yaml", "Path to the manifest to send to the RP")
+		configBlob = flag.String("configBlob", "../../_data/containerservice.yaml", "Path to the OpenShift internal config blob")
+	)
+
+	BeforeEach(func() {
+		var err error
+		cli, err = azure.NewClientFromEnvironment()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should be possible to maintain a healthy cluster after rotating all credentials", func() {
+		ctx := context.Background()
+		ctx = context.WithValue(ctx, api.ContextKeyClientID, os.Getenv("AZURE_CLIENT_ID"))
+		ctx = context.WithValue(ctx, api.ContextKeyClientSecret, os.Getenv("AZURE_CLIENT_SECRET"))
+		ctx = context.WithValue(ctx, api.ContextKeyTenantID, os.Getenv("AZURE_TENANT_ID"))
+
+		By("Parsing the external manifest")
+		external, err := parseExternalConfig(*manifest)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(external).NotTo(BeNil())
+
+		By("Parsing the internal manifest containing config blob")
+		internal, err := managedcluster.ReadConfig(*configBlob)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(internal).NotTo(BeNil())
+
+		By("Deleting all non-ca cluster certificates and credentials from the config blob...")
+		mutated := deleteSecrets(internal)
+		Expect(mutated).NotTo(BeNil())
+
+		By("Running generate on the modified config blob")
+		pluginConfig, err := fakerp.GetPluginConfig()
+		Expect(err).NotTo(HaveOccurred())
+		configGen := config.NewSimpleGenerator(pluginConfig)
+		Expect(configGen).NotTo(BeNil())
+		err = configGen.Generate(mutated)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Persisting the config blob containing the new certificates and credentials")
+		err = saveConfig(mutated, *configBlob)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Calling update on the fake rp with the updated config blob")
+		updated, err := cli.UpdateOSACluster(ctx, external, pluginConfig)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(updated).NotTo(BeNil())
+
+		By("Parsing the config blob after the update")
+		internalAfterUpdate, err := managedcluster.ReadConfig(*configBlob)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(internalAfterUpdate).NotTo(BeNil())
+
+		By("Verifying that the initial config blob does not match the one created after the update")
+		configMatch := reflect.DeepEqual(internal.Config.Certificates, internalAfterUpdate.Config.Certificates)
+		Expect(configMatch).To(BeFalse())
+
+		By("Verifying that the mutated config blob matches the one created after the update")
+		configMatch = reflect.DeepEqual(mutated.Config.Certificates, internalAfterUpdate.Config.Certificates)
+		Expect(configMatch).To(BeTrue())
+	})
+})
+
+// deleteSecrets removes all non-ca certificates, private keys and secrets from OpenShiftManagedCluster.Config
+func deleteSecrets(config *api.OpenShiftManagedCluster) *api.OpenShiftManagedCluster {
```

Comment:
This code should move into a newly created function inside the plugin, which would be listed in the external plugin interface, which would be responsible for resetting these values and calling Update on the cluster.  You'd call into that from here via the admin API, which would call that function.  This work should also be part of the follow-up mentioned above.

Reply:
here #807